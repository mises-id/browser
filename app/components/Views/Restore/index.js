/*
 * @Author: lmk
 * @Date: 2021-07-19 12:17:48
 * @LastEditTime: 2021-07-19 16:24:15
 * @LastEditors: lmk
 * @Description: Restore misesid
 */
import { strings } from 'app/locales/i18n';
import React, { useReducer, useEffect, useRef, useState } from 'react';
import {Platform, StyleSheet, Text, View,Image ,TouchableOpacity, TextInput, Button} from 'react-native';
const word = (item={},key)=>{
  return <View key={key} style={styles.wordContent}>
    <Text style={styles.wordTxt}>{item.value}</Text>
  </View>
}
const Restore = (props)=>{
  const [data, setdata] = useState([{value:'cloud'},{value:'rabbit'},{value:'fly'},{value:'hawk'},{value:'contemplate'},{value:'pretty'},{value:'superduty'},{value:'lovely'},{value:'come'},{value:'gift'},{value:'main'},{value:'green'}])
  return <View style={styles.pageBox}>
    <View style={styles.titleBox}><Text style={styles.title}>{strings('restore.title')}</Text></View>
    <View style={styles.inputBox}>
      <TextInput placeholder={strings('common.placeholder')}></TextInput>
    </View>
    <View style={styles.wordsContainer}>
      {data.map(word)}
    </View>
    <View style={styles.btnBox}>
      <View style={[styles.btnStyle,styles.success]}>
        <Text style={styles.successBtnTxt}>{strings('restore.success_button')}</Text>
      </View>
      <View style={[styles.btnStyle,styles.cancel]}>
        <Text style={styles.cancelBtnTxt}>{strings('common.cancel_button')}</Text>
      </View>
    </View>
  </View>
}
const styles = StyleSheet.create({
  pageBox:{
    paddingLeft:15,
    paddingRight:15
  },
  title:{
    fontSize:16,
    color:"#333",
    fontWeight:"bold"
  },
  titleBox:{
    paddingTop:25,
    paddingBottom:17
  },
  inputBox:{
    height:50,
    backgroundColor:"#F8F8F8",
    borderRadius:5,
    paddingLeft:15,
    paddingRight:15,
    marginBottom:25
  },
  wordsContainer:{
    borderColor:"#EEEEEE",
    borderRadius:5,
    borderWidth:1,
    flexDirection: 'row',
    flexWrap:"wrap",
    padding:5
  },
  wordContent:{
    padding:15,
    borderRadius:5,
    position:'relative',
    alignItems:'center',
    justifyContent:'center',
    backgroundColor:'#F7F7FF',
    margin:5,
  },
  wordTxt:{
    fontSize:16,
    fontWeight:'bold',
    color:'#5D61FF'
  },
  btnBox:{
    marginTop:15,
    marginLeft:25,
    marginRight:25
  },
  btnStyle:{
    borderWidth:1,
    height:50,
    borderRadius:50,
    alignItems:'center',
    justifyContent:'center',
    overflow:'hidden',
    marginTop:25
  },
  success:{
    borderColor:'#5D61FF'
  },
  successBtnTxt:{
    fontSize:16,
    color:"#5D61FF",
    fontWeight:'bold'
  },
  cancel:{
    borderColor:'#EEEEEE',
    backgroundColor:'#EEEEEE'
  },
  cancelBtnTxt:{
    fontSize:16,
    color:"#666",
    fontWeight:'bold'
  }
})
Restore.navigationOptions = ()=>{
  return {
    headerTitle:strings('menu.restore')
  }
}
export default Restore