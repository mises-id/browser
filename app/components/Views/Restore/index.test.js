/*
 * @Author: lmk
 * @Date: 2021-07-19 22:12:23
 * @LastEditTime: 2021-07-19 22:13:02
 * @LastEditors: lmk
 * @Description: Restore test
 */
import React from 'react';
import {shallow} from 'enzyme';
import Restore from './';

describe('Restore', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Restore/>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
