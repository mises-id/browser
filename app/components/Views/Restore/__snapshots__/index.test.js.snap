// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restore should render correctly 1`] = `
<View
  style={
    Object {
      "paddingLeft": 15,
      "paddingRight": 15,
    }
  }
>
  <View
    style={
      Object {
        "paddingBottom": 17,
        "paddingTop": 25,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Enter the mnemonic words in order
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderRadius": 5,
        "height": 50,
        "marginBottom": 25,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <TextInput
      onChange={[Function]}
      placeholder="Please enter the contents"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": 25,
        "marginRight": 25,
        "marginTop": 15,
      }
    }
  >
    <TouchableOpacity
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 50,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 25,
              "overflow": "hidden",
            },
            Object {
              "borderColor": "#bdbfff",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#bdbfff",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Restore success
        </Text>
      </View>
    </TouchableOpacity>
    <TouchableOpacity
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 50,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 25,
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#EEEEEE",
              "borderColor": "#EEEEEE",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#666",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Cancel
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
